### PR TITLE
Feat(script): mac binary release

### PR DESCRIPTION
**Pull request by ailisp**
Add mac binary release and release to S3

Test Plan
------------
binary should be build and work when download to a local mac

### CODE REVIEW

**Comment by ailisp** [comment]
It's weird travis isn't triggered :thinking:

**Comment by ailisp** [comment]
This is still WIP. move from draft because it was not trigger travis

**Comment by ailisp** [comment]
@bowenwang1996 use travis mac to build mac binary. However it seems too slow:

image

**Comment by ailisp** [comment]
travis has a hard timeout 50 minute https://docs.travis-ci.com/user/customizing-the-build#build-timeouts. Seems github action has a mac instance type too, will try binary build on that

**Comment by ailisp** [comment]
First success build: https://s3-us-west-1.amazonaws.com/build.nearprotocol.com/nearcore/Darwin/mac-binary-release/near
would like to know if this work on your Mac

File: scripts/binary-release.sh [hunk 1]
```diff
@@ -1,8 +1,8 @@
 #!/bin/bash
 set -euo pipefail
 
-branch=${BUILDKITE_BRANCH}
-commit=${BUILDKITE_COMMIT}
+branch=${BUILDKITE_BRANCH:-${GITHUB_REF##*/}}
```
Comment (bowenwang1996) [inline scripts/binary-release.sh:4]: what is this magic?

**Comment by ilblackdragon** [comment]
@ailisp attached binary works.

File: scripts/binary-release.sh (hunk 1 above)
Comment (ailisp) [reply scripts/binary-release.sh:4]: steals here: https://stackoverflow.com/questions/58033366/how-to-get-current-branch-within-github-actions
github actions doesn't give you the branch as env var

**Pull request merged**